### PR TITLE
feat: add navbar to support multi language

### LINKED ISSUE #16: Feature: Translate the course to other language
### Type of feature

📝 Documentation

### Current behavior

Hi, it is a perfect course for beginners and I would like to introduce it to my newborn open-source community in Angola(github.com/angolaosc), but the biggest part of my community do not speak or understand English. 

### Suggested solution

It would be perfect if we can have a translation of this course.

I will be available to contribute translating most of the part of this course into Portuguese and encourage people from my community to contribute too.

### Additional context

idiom

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by geoffreylgv**
## Description
This PR is a feature adding a sidebar for the multi-language version of the intro course.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #16 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: _layouts/navbar.md [hunk 1]
```diff
@@ -1,2 +1,5 @@
 * [Home Page](/)
-* [⭐ Star The Repo](https://github.com/open-sauced/intro)
\ No newline at end of file
+* [⭐ Star The Repo](https://github.com/open-sauced/intro)
+* Languages
+    * [:us: English](/)
+    * [:fr: French](/translations/fr/)
```
Comment (antonio-pedro99) [inline _layouts/navbar.md:5]: Let's keep this line out of this PR, as this PR should not be specific to some idiom but English.

Comment (antonio-pedro99) [reply _layouts/navbar.md:5]: You can have this line in #38

File: translations/_layouts/navbar.md
Comment (antonio-pedro99) [inline translations/_layouts/navbar.md:1]: This is a French-related stuff, let's move it to #38 only.
Again, this PR should not focus on any translation.

File: translations/_layouts/sidebar.md
Comment (antonio-pedro99) [inline translations/_layouts/sidebar.md:1]: please read my previous comment

File: translations/fr/README.md
Comment (antonio-pedro99) [inline translations/fr/README.md:1]: please read my previous comment

**Review by antonio-pedro99 — CHANGES_REQUESTED** [review]
Nice job @geoffreylgv.
Apart from my specific comments, here is one of the reasons to not including French related translation in this PR.

- This PR introduces the basic structure to support multiple language. 
- Once merged, to add a new idiom translation, I would just add the language in the `_layouts/navbar.md` and create a new folder under translations with the language short form, like `pt-br` for Brazilian Portuguese.

This is to avoid conflicts with #38 and #35. 
And also, you can not add semi translated docs here. If this PR gets merged now, users will see that there's a French version, but they won't see because it is not available yet. :)

**Comment by geoffreylgv** [comment]
> Nice job @geoffreylgv. Apart from my specific comments, here is one of the reasons to not including French related translation in this PR.
> 
> * This PR introduces the basic structure to support multiple language.
> * Once merged, to add a new idiom translation, I would just add the language in the `_layouts/navbar.md` and create a new folder under translations with the language short form, like `pt-br` for Brazilian Portuguese.
> 
> This is to avoid conflicts with #38 and #35. And also, you can not add semi translated docs here. If this PR gets merged now, users will see that there's a French version, but they won't see because it is not available yet. :)
> 
> IMHO, the only change we need in this PR is:
> 
> ```
> * Languages
>     * [:us: English](/)
> ```
> 
> The rest you can include in #38.

Hello @antonio-pedro99 
I've removed french related contents to have a single temple for the navbar
This means in the `translations/country-code/`, we should have `_layouts/sidebar.md` and `_layouts/navbar.md` to have the navbar and sidebar in the specific language we are translating

File: index.html [hunk 2]
```diff
@@ -10,7 +10,7 @@
       name="viewport"
       content="width=device-width, initial-scale=1.0, minimum-scale=1.0"
     />
-    <link rel="stylesheet" href="./styles/custom.css" />
+    <!-- <link rel="stylesheet" href="./styles/custom.css" /> -->
```
Comment (antonio-pedro99) [inline index.html:13]: @geoffreylgv why did you comment this line?

**Review by antonio-pedro99 — APPROVED** [review]
LGTM
Left a comment.

**Comment by BekahHW** [comment]
When I'm in dark mode, the text for the language gets lighter. I think this is a docsify thing, but let me know if there's a different setting we need to add.

Also, do you know why the dropdown with the language disappears if you don't comment out 	`    
`

File: index.html (hunk 2 above)
Comment (CBID2) [reply index.html:13]: We found that the drop-down appears when it’s commented @antonio-pedro99

**Comment by geoffreylgv** [comment]
> When I'm in dark mode, the text for the language gets lighter. I think this is a docsify thing, but let me know if there's a different setting we need to add.
> 
Hi @BekahHW, for the dark/light mode on language dropdown, it's a docsify thing; let dig on settings that can help to have it also changeable.
> 
> Also, do you know why the dropdown with the language disappears if you don't comment out ` <!-- <link rel="stylesheet" href="./styles/custom.css" /> -->`

And for the CSS commented, hmm there is a way actually to have the centered navbar behavior

**Comment by BekahHW** [comment]
@geoffreylgv for the commented out portion, does that impact css elsewhere? I haven't tested it out, but before merging this in, I want to make sure that it's not impacting things I haven't seen

**Comment by geoffreylgv** [comment]
> @geoffreylgv for the commented out portion, does that impact css elsewhere? I haven't tested it out, but before merging this in, I want to make sure that it's not impacting things I haven't seen

No it doesn't affect, @BekahHW 
In the meantime I'll check the CSS and propose a minor change (maybe write one line or edit twoo in the custom style) if we should keep the CSS line not commented.

**Comment by geoffreylgv** [comment]
> @geoffreylgv for the commented out portion, does that impact css elsewhere? I haven't tested it out, but before merging this in, I want to make sure that it's not impacting things I haven't seen

Hey @CBID2 @BekahHW @antonio-pedro99 
I will kindly ask to check this behavior on your end

By editing the position property from `unset` to `relative`
~`position: unset !important;`~
`position: relative !important;`

Final app-nav style

```bash
.app-nav {
	display: flex;
	align-items: center;
	justify-content: center;
	position: relative !important;
	margin-top: 16px !important;
	margin-right: unset !important;
}
```

Basically, this means to keep the `<link rel="stylesheet" href="./styles/custom.css" />` uncommented in the html and to edit the `custom.css` file
The `unset` property tells to ignore any inherited position property on the `app-nav` who is `static` by default

**Comment by CBID2** [comment]
> > @geoffreylgv for the commented out portion, does that impact css elsewhere? I haven't tested it out, but before merging this in, I want to make sure that it's not impacting things I haven't seen
> 
> Hey @CBID2 @BekahHW @antonio-pedro99 I will kindly ask to check this behavior on your end
> 
> By editing the position property from `unset` to `relative` ~`position: unset !important;`~ `position: relative !important;`
> 
> Final app-nav style
> 
> ```shell
> .app-nav {
> 	display: flex;
> 	align-items: center;
> 	justify-content: center;
> 	position: relative !important;
> 	margin-top: 16px !important;
> 	margin-right: unset !important;
> }
> ```
> 
> Basically, this means to keep the `<link rel="stylesheet" href="./styles/custom.css" />` uncommented in the html and to edit the `custom.css` file The `unset` property tells to ignore any inherited position property on the `app-nav` who is `static` by default

Hey @geoffreylgv. I can't test it locally unless you grant me permission to push code to your PR.

**Comment by geoffreylgv** [comment]
> > > @geoffreylgv for the commented out portion, does that impact css elsewhere? I haven't tested it out, but before merging this in, I want to make sure that it's not impacting things I haven't seen
> > 
> > 
> > Hey @CBID2 @BekahHW @antonio-pedro99 I will kindly ask to check this behavior on your end
> > By editing the position property from `unset` to `relative` ~`position: unset !important;`~ `position: relative !important;`
> > Final app-nav style
> > ```shell
> > .app-nav {
> > 	display: flex;
> > 	align-items: center;
> > 	justify-content: center;
> > 	position: relative !important;
> > 	margin-top: 16px !important;
> > 	margin-right: unset !important;
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Basically, this means to keep the `<link rel="stylesheet" href="./styles/custom.css" />` uncommented in the html and to edit the `custom.css` file The `unset` property tells to ignore any inherited position property on the `app-nav` who is `static` by default
> 
> Hey @geoffreylgv. I can't test it locally unless you grant me permission to push code to your PR. Invite me to your repo

Ok @CBID2 done! Check your mail

**Comment by CBID2** [comment]
> > > > @geoffreylgv for the commented out portion, does that impact css elsewhere? I haven't tested it out, but before merging this in, I want to make sure that it's not impacting things I haven't seen
> > > 
> > > 
> > > Hey @CBID2 @BekahHW @antonio-pedro99 I will kindly ask to check this behavior on your end
> > > By editing the position property from `unset` to `relative` ~`position: unset !important;`~ `position: relative !important;`
> > > Final app-nav style
> > > ```shell
> > > .app-nav {
> > > 	display: flex;
> > > 	align-items: center;
> > > 	justify-content: center;
> > > 	position: relative !important;
> > > 	margin-top: 16px !important;
> > > 	margin-right: unset !important;
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Basically, this means to keep the `<link rel="stylesheet" href="./styles/custom.css" />` uncommented in the html and to edit the `custom.css` file The `unset` property tells to ignore any inherited position property on the `app-nav` who is `static` by default
> > 
> > 
> > Hey @geoffreylgv. I can't test it locally unless you grant me permission to push code to your PR. Invite me to your repo
> 
> Ok @CBID2 done! Check your mail

Thanks @geoffreylgv. I pushed the code. The text does come out light in dark mode, but once I hover over the language name, it darkens a bit. 

[testing-navbar](https://github.com/open-sauced/intro/assets/105683440/cedcaeed-20b7-4c6b-9a26-2ddc16db1ddf)

**Comment by BekahHW** [comment]
lgtm!

**Comment by antonio-pedro99** [comment]
Well done @geoffreylgv

**Comment by CBID2** [comment]
Woohoo! We did @geoffreylgv and @antonio-pedro99! Now onto PR #38 @BekahHW!

**Comment by geoffreylgv** [comment]
> Well done @geoffreylgv
> Woohoo! We did @geoffreylgv and @antonio-pedro99! Now onto PR https://github.com/open-sauced/intro/pull/38 @BekahHW!
>
Thank you all, we did it😃

**Pull request merged**